src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -413,16 +442,83 @@ public void run() {
       holder.createErrorAnnotation(currentModifiers.getModifierPsi(HaxeModifierType.OVERRIDE), "Overriding nothing").registerFix(
         new RemoveModifierIntent("Remove override", currentModifiers, HaxeModifierType.OVERRIDE)
       );
-    } else if (!currentModifiers.hasModifier(HaxeModifierType.OVERRIDE) && requiredOverride) {
-      holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), "Must override").registerFix(
-        new HaxeFixer("Add override") {
-          @Override
-          public void run() {
-            currentModifiers.addModifier(HaxeModifierType.OVERRIDE);
+    }
+    else if (requiredOverride) {
+      if (!currentModifiers.hasModifier(HaxeModifierType.OVERRIDE)) {
+        holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), "Must override").registerFix(
+          new HaxeFixer("Add override") {
+            @Override
+            public void run() {
+              currentModifiers.addModifier(HaxeModifierType.OVERRIDE);
+            }
           }
-        }
+        );
+      }
+      else {
+        // It is rightly overriden. Now check the signature.
+        checkMethodsSignatureCompatibility(currentMethod, parentMethod, holder);
+      }
+    }
+  }
+
+  static public void checkMethodsSignatureCompatibility(
+    @NotNull final HaxeMethodModel currentMethod,
+    @NotNull final HaxeMethodModel parentMethod,
+    final AnnotationHolder holder
+  ) {
+    final HaxeDocumentModel document = currentMethod.getDocument();
```

Comment:
checkMethodsSignatureCompatibility doesn't check function calls, but function definitions (to check overrides and implementations). Not sure if haxe allows you to skip optional arguments. I didn't know that it allowed to have optional arguments in any part of the signature, and call methods as you put in the example, that's a bit odd.

Method calls checks are not in SemanticAnalyzer class but, in order to avoid duplicated code, they are part of the expression evaluator which accepts a context to emit errors.

But i would treat oddities like that one in other pull request, providing false-positive and missing errors as tests to avoid regressions.